feat: Add Reset Password button and functionality in User tab

thephani opened the pull request:
## Describe your changes
`User Tab` [popup] The “Reset Password” option in the User tab can be enabled or disabled using the options.

## Issue ticket number and link
https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/486

## Checklist before requesting a review
- [X] I have read and understand the [Contributions section](https://github.com/tprouvot/Salesforce-Inspector-reloaded#contributions)
- [X] Target branch is releaseCandidate and not master
- [X] I have performed a self-review of my code
- [X] I ran the [unit tests](https://github.com/tprouvot/Salesforce-Inspector-reloaded#unit-tests) and my PR does not break any tests
- [X] I documented the changes I've made on the [CHANGES.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/CHANGES.md) and followed actual conventions
- [ ] I added a new section on [how-to.md](https://github.com/tprouvot/Salesforce-Inspector-reloaded/blob/master/docs/how-to.md) (optional)

Commit:
feat: Add Reset Password button and functionality in User tab

Commit:
improved msg

tprouvot commented on addon/popup.js:
```diff
@@ -3242,6 +3242,22 @@ class UserDetails extends React.PureComponent {
+        "/services/data/v" + apiVersion + "/sobjects/User/" + user.Id + "/password",
+        {method: "DELETE"}
+      );
+      this.showSuccessToast(
```
Could you check for the api call response and handle the case when INSUFFICIENT_ACCESS error is returned ?

https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_user_password_delete.htm

thephani replied:
@tprouvot am handling it like this
<img width="528" height="254" alt="image" src="https://github.com/user-attachments/assets/cd43feae-83bd-48f3-8e0f-f772e9426a75" />

do you want me to add the `INSUFFICIENT_ACCESS ` msg in the bannder?

tprouvot commented on CHANGES.md:
```diff
@@ -5,6 +5,7 @@
 - `Popup` Add "Copy Id" icon to User tab [feature 906](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/906) (contribution by [Arnav Gupta](https://github.com/ArnavGuptaaa)).
 - `Data Export` Auto-populate SOQL editor from Salesforce List View context [feature 969](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/969) (request and contribution by [Temam Abdelkrim](https://github.com/KRIMOD)) and [feature 817](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/817) (request by [Nicolas Vuillamy](https://github.com/nvuillam))
 - `Data Export` Prevent popup actions menu for Aggregate result [feature #663](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/663) (contribution by [Phanidra Mangipudi](https://github.com/thephani))
+- `User Tab` [popup] The “Reset Password” option in the User tab can be enabled or disabled using the options. [feature #486](https://github.com/tprouvot/Salesforce-Inspector-reloaded/issues/486) (contribution by [Phanidra Mangipudi](https://github.com/thephani))
```
Could you add this new addition on the top of the list ?

tprouvot reviewed (changes requested):
Hi @thephani 
Thanks for the PR ! 

Here are small changes to do, also could you document this new feature in the how-to.md file ?

Commit:
Refactor code structure for improved readability and maintainability

thephani committed:
Revise User Tab Reset Password button section

Updated the section on the Reset Password button for clarity and improved formatting.

thephani committed:
Merge branch 'releaseCandidate' into feat/486-reset-pwd

tprouvot committed:
Hide button for current user, resolve display issue in option for small screens, simplify button label and toast messages

tprouvot committed:
Merge branch 'releaseCandidate' into feat/486-reset-pwd
